src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,407 @@
+import java.util.Optional;
+import java.util.Map;
+
+public class AzureDevOpsServerPullRequestDecorator implements PullRequestBuildStatusDecorator {
+
+    // SCANNER PROPERTIES
+    public static final String PULLREQUEST_AZUREDEVOPS_API_VERSION = "sonar.pullrequest.vsts.apiVersion";     // sonar.pullrequest.vsts.apiVersion=5.1-preview.1
+    public static final String PULLREQUEST_AZUREDEVOPS_INSTANCE_URL = "sonar.pullrequest.vsts.instanceUrl";   // sonar.pullrequest.vsts.instanceUrl=https://dev.azure.com/fabrikam/
+    public static final String PULLREQUEST_AZUREDEVOPS_BASE_BRANCH = "sonar.pullrequest.base";                // sonar.pullrequest.base=master
+    public static final String PULLREQUEST_AZUREDEVOPS_BRANCH = "sonar.pullrequest.branch";                   // sonar.pullrequest.branch=feature/some-feature
+    public static final String PULLREQUEST_AZUREDEVOPS_PULLREQUEST_ID = "sonar.pullrequest.key";              // sonar.pullrequest.key=222
+    public static final String PULLREQUEST_AZUREDEVOPS_PROJECT_ID = "sonar.pullrequest.vsts.project";         // sonar.pullrequest.vsts.project=MyProject
+    public static final String PULLREQUEST_AZUREDEVOPS_REPOSITORY_NAME = "sonar.pullrequest.vsts.repository"; // sonar.pullrequest.vsts.repository=MyReposytory
+
+    // AZURE DEVOPS ENVIRONMENT VARIABLES: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
+    public static final String AZUREDEVOPS_ENV_INSTANCE_URL = "System.TeamFoundationCollectionUri";
+    public static final String AZUREDEVOPS_ENV_BASE_BRANCH = "System.PullRequest.TargetBranch";
+    public static final String AZUREDEVOPS_ENV_BRANCH = "System.PullRequest.SourceBranch";
+    public static final String AZUREDEVOPS_ENV_PULLREQUEST_ID = "System.PullRequest.PullRequestId";
+    public static final String AZUREDEVOPS_ENV_TEAMPROJECT_ID = "System.TeamProjectId";
+    public static final String AZUREDEVOPS_ENV_REPOSITORY_NAME = "Build.Repository.Name";
+    
+    public static final String API_VERSION_PREFIX = "?api-version=";
+    
+    private static final Logger LOGGER = Loggers.get(AzureDevOpsServerPullRequestDecorator.class);
+    
+    private final Server server;
+    private final ScmInfoRepository scmInfoRepository;
+        
+    public AzureDevOpsServerPullRequestDecorator(Server server, ScmInfoRepository scmInfoRepository) {
+        super();
+        this.server = server;
+        this.scmInfoRepository = scmInfoRepository;
+    }
+
+    @Override
+    public DecorationResult decorateQualityGateStatus(AnalysisDetails analysisDetails, AlmSettingDto almSettingDto, ProjectAlmSettingDto projectAlmSettingDto) {
+        
+        String apiVersion = "6.0-preview.1";
+        String authorizationHeader;        
+        String azureUrl = "";
+        String baseBranch = "";
+        String branch = "";
+        String pullRequestId = "";
+        String azureRepositoryName = "";
+        String azureProjectId = "";
```

Comment:
Are these all use `Optional.orElse` during initialisation, there's no need to pre-initialise them here. I think you could possibly move their definition to the point you first try and initialise them, but could just leave their values as undefined here otherwise.

Reply:
Fixed